src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LevelCrossingOnRailwayCheck.java

```diff
@@ -0,0 +1,369 @@
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        /*-
+         * The following objects should be checked:
+         *  1) Any node.
+         *  2) Any object that is tagged with railway=level_crossing.
+         *  3) Any object that is tagged as a railway as indicted in railway.filter.
+         */
+        return object instanceof Node
+                || Validators.isOfType(object, RailwayTag.class, RailwayTag.LEVEL_CROSSING)
+                || this.railwayFilter.test(object);
+    }
+
+    /**
+     * Create a Return Flags for level_crossing objects.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that contains flagged issue details
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        /*-
+         * The following invalid situations are to be flagged:
+         *  1) object is node and
+         *     a) is marked as a level crossing but does in not an intersection of highway and railway
+         *     b) is not tagged as a level crossing and is an intersection of highway and railway.
+         *  2) object is not a node or point and is tagged with railway=level_crossing.
+         *  3) object is railway and intersects a highway on the same layer but there is no node.
+         */
+
+        final Optional<CheckFlag> flagIncorrectlyTagged = this.flagIncorrectlyTagged(object);
+        if (!flagIncorrectlyTagged.isEmpty())
+        {
+            return flagIncorrectlyTagged;
+        }
+        final Optional<CheckFlag> flagNonNodeTagged = this.flagNonNodeTagged(object);
+        if (!flagNonNodeTagged.isEmpty())
+        {
+            return flagNonNodeTagged;
+        }
+        final Optional<CheckFlag> flagInvalidIntersections = this.flagInvalidIntersections(object);
+        if (!flagInvalidIntersections.isEmpty())
+        {
+            return flagInvalidIntersections;
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Flag nodes incorrectly tagged with level_crossing or missing level_crossing tag.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that contains flagged issue details
+     */
+    private Optional<CheckFlag> flagIncorrectlyTagged(final AtlasObject object)
+    {
+        if (object instanceof Node)
+        {
+            final Node node = (Node) object;
+
+            final int issueType = this.isValidLevelCrossingNode(node);
+            if (Validators.isOfType(node, RailwayTag.class, RailwayTag.LEVEL_CROSSING)
+                    && issueType > 0)
```

Comment:
nit: `issueType` could be better served as an Enum to avoid referencing docs for explanation

Reply:
Addressed in latest version.